src/fsharp/LexFilter.fs

```diff
@@ -1632,14 +1700,14 @@ type LexFilterImpl (lightStatus: LightSyntaxStatus, compilingFsLib, lexer, lexbu
         //  ...
         // ...
 
-        | _, CtxtInterfaceHead offsidePos :: _ 
-                when isSemiSemi || (if isInterfaceContinuator token then tokenStartCol + 1 else tokenStartCol) <= offsidePos.Column -> 
+        | _, CtxtInterfaceHead offsidePos :: _
+                when isSemiSemi || (if relaxWhitespace2OffsideRule || isInterfaceContinuator token then tokenStartCol + 1 else tokenStartCol) <= offsidePos.Column -> 
             if debug then dprintf "token at column %d is offside from INTERFACE(offsidePos=%a)! pop and reprocess\n" tokenStartCol outputPos offsidePos
             popCtxt()
             reprocess()
 
-        | _, CtxtTypeDefns offsidePos :: _ 
-                when isSemiSemi || (if isTypeContinuator token then tokenStartCol + 1 else tokenStartCol) <= offsidePos.Column -> 
+        | _, CtxtTypeDefns offsidePos :: _
+                when isSemiSemi || (if relaxWhitespace2OffsideRule || isTypeContinuator token then tokenStartCol + 1 else tokenStartCol) <= offsidePos.Column -> 
```

Comment:
I'm getting the hang of the implementation. It would be great if for each of theses changes you could put a note here in the PR of a link through to testing that corresponds to the cahnge.